server/schedule/filter/filters.go

```diff
@@ -228,67 +228,80 @@ type StoreStateFilter struct {
 	MoveRegion bool
 	// Set true if allows temporary states.
 	AllowTemporaryStates bool
+	// Reason is used to distinguish the reason of store state filter
+	Reason string
 }
 
 // Scope returns the scheduler or the checker which the filter acts on.
-func (f StoreStateFilter) Scope() string {
+func (f *StoreStateFilter) Scope() string {
 	return f.ActionScope
 }
 
 // Type returns the type of the Filter.
-func (f StoreStateFilter) Type() string {
-	return "store-state-filter"
+func (f *StoreStateFilter) Type() string {
+	return fmt.Sprintf("store-state-%s-filter", f.Reason)
 }
 
 // conditionFunc defines condition to determine a store should be selected.
 // It should consider if the filter allows temporary states.
 type conditionFunc func(*config.PersistOptions, *core.StoreInfo) bool
 
-func (f StoreStateFilter) isTombstone(opt *config.PersistOptions, store *core.StoreInfo) bool {
+func (f *StoreStateFilter) isTombstone(opt *config.PersistOptions, store *core.StoreInfo) bool {
+	f.Reason = "tombstone"
 	return store.IsTombstone()
 }
 
-func (f StoreStateFilter) isDown(opt *config.PersistOptions, store *core.StoreInfo) bool {
+func (f *StoreStateFilter) isDown(opt *config.PersistOptions, store *core.StoreInfo) bool {
+	f.Reason = "down"
 	return store.DownTime() > opt.GetMaxStoreDownTime()
 }
 
-func (f StoreStateFilter) isOffline(opt *config.PersistOptions, store *core.StoreInfo) bool {
+func (f *StoreStateFilter) isOffline(opt *config.PersistOptions, store *core.StoreInfo) bool {
+	f.Reason = "offline"
 	return store.IsOffline()
 }
 
-func (f StoreStateFilter) pauseLeaderTransfer(opt *config.PersistOptions, store *core.StoreInfo) bool {
+func (f *StoreStateFilter) pauseLeaderTransfer(opt *config.PersistOptions, store *core.StoreInfo) bool {
+	f.Reason = "pause-leader"
 	return !store.AllowLeaderTransfer()
 }
 
-func (f StoreStateFilter) isDisconnected(opt *config.PersistOptions, store *core.StoreInfo) bool {
+func (f *StoreStateFilter) isDisconnected(opt *config.PersistOptions, store *core.StoreInfo) bool {
+	f.Reason = "disconnected"
 	return !f.AllowTemporaryStates && store.IsDisconnected()
 }
 
-func (f StoreStateFilter) isBusy(opt *config.PersistOptions, store *core.StoreInfo) bool {
+func (f *StoreStateFilter) isBusy(opt *config.PersistOptions, store *core.StoreInfo) bool {
+	f.Reason = "busy"
 	return !f.AllowTemporaryStates && store.IsBusy()
 }
 
-func (f StoreStateFilter) exceedRemoveLimit(opt *config.PersistOptions, store *core.StoreInfo) bool {
+func (f *StoreStateFilter) exceedRemoveLimit(opt *config.PersistOptions, store *core.StoreInfo) bool {
+	f.Reason = "remove-limit"
 	return !f.AllowTemporaryStates && !store.IsAvailable(storelimit.RemovePeer)
 }
 
-func (f StoreStateFilter) exceedAddLimit(opt *config.PersistOptions, store *core.StoreInfo) bool {
+func (f *StoreStateFilter) exceedAddLimit(opt *config.PersistOptions, store *core.StoreInfo) bool {
+	f.Reason = "add-limit"
```

Comment:
what about `exceed-limit` here. it would be mor clear.

Reply:
Do you mean exceed-add-limit?